src/email/sender.js

```diff
@@ -0,0 +1,36 @@
+const nodemailer = require('nodemailer');
+const path = require('path');
+const ejs = require('ejs');
+require('dotenv').config();
+
+async function sendForgotPasswordEmail(toEmail, resetToken) {
+  const transporter = nodemailer.createTransport({
+    service: 'gmail',
+    auth: {
+      user: process.env.EMAIL_USER,
+      pass: process.env.EMAIL_PASSWORD,
+    },
+  });
+
+  // Load and render the HTML template
+  const htmlTemplate = await ejs.renderFile(
+    path.join(__dirname, '../email/templates/html.ejs'),
+    { resetLink: `http://localhost:8000/api/v1/reset-password/${resetToken}` }
+  );
+
+  const mailOptions = {
+    from: 'your@example.com',
+    to: toEmail,
+    subject: 'Password Reset Request',
+    html: htmlTemplate,
+  };
+
+  try {
+    await transporter.sendMail(mailOptions);
+    console.log('Password reset email sent successfully');
+  } catch (error) {
+    console.error('Error sending password reset email:', error);
+  }
+}
+
+module.exports = sendForgotPasswordEmail;
```

Comment:
Right now we're making the default value of this export `sendForgotPasswordEmail` function. 

That means if you import it in a different file:  
 

`const emailSender = require('./email/sender')`

Then the value of `emailSender` is actually just the function `sendForgotPasswordEmail`. 

We will eventually be adding other sending functions to this file so we shouldn't make this just a single value, but instead export an object, so that the other future functions can be exported as well:

```suggestion
module.exports = { sendForgotPasswordEmail };
```

Then this can be used in other files like this:

`const emailSender = require('./email/sender')` (where `emailSender` will now be an object. And you can call this forgot password function by grabbing the value at this key: `emailSender.sendForgotPasswordEmail(...)`)

or like this:

`const {sendForgotPasswordEmail} = require('./email/sender')` . Here we're using object destructuring to get just the field we want out of the object. Then we can use like this: `sendForgotPasswordEmail(...)`